cranelift/filetests/src/test_zkasm.rs

```diff
@@ -237,6 +240,62 @@ mod tests {
         program.join("\n")
     }
 
+    fn run_wat_file(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+        let engine = Engine::default();
+        let wat = std::fs::read_to_string(path).unwrap();
+        let pat = r#"(import "env" "assert_eq" (func $assert_eq (param i32) (param i32)))"#;
+        let assert_type = if wat.contains(pat) { 32 } else { 64 };
+        let binary = wat::parse_file(path)?;
+        let module = Module::new(&engine, &binary)?;
+        if assert_type == 64 {
```

Comment:
It looks like only `host_func` is different in those branches, which might allow to simplify it to:

```rust
let module = ...;
let store = ...;
let host_function = match {
	64 => Func::wrap(...);
	32 => Func::wrap(...);
	_ => panic!(...);
}
let _instance = ...;
```

Reply:
It's also difference in ``store``. In one case it is ``Store<i32>`` in the other -- ``Store<i64>``